MapboxCoreNavigation/RouteController.swift

```diff
@@ -334,19 +392,22 @@ extension RouteController: CLLocationManagerDelegate {
         updateVisualInstructionProgress()
 
         guard userIsOnRoute(location) || !(delegate?.routeController?(self, shouldRerouteFrom: location) ?? true) else {
-            reroute(from: location, along: routeProgress)
+            rerouteForDiversion(from: location, along: routeProgress)
             return
         }
 
         updateSpokenInstructionProgress()
 
         // Check for faster route given users current location
         guard reroutesProactively else { return }
-        // Only check for faster alternatives if the user has plenty of time left on the route.
+        
+        // Only check for faster routes or ETA updates if the user has plenty of time left on the route (10+min)
         guard routeProgress.durationRemaining > 600 else { return }
-        // If the user is approaching a maneuver, don't check for a faster alternatives
+        
+        // If the user is approaching a maneuver (within 70secs of the maneuver), don't check for a faster routes or ETA updates
         guard routeProgress.currentLegProgress.currentStepProgress.durationRemaining > RouteControllerMediumAlertInterval else { return }
```

Comment:
Also, why no ETA updates in this case? Again, for reroutes it makes sense, but I don't get why you wouldn't want to update ETA.

Reply:
The route differs a maximum of 10%. That could differ theoretically in an instruction in the beginning of the new route so the user has to suddenly change direction. I've kept these checks to keep the change to a minimum before further examining which parts we can optimise to deliver a better experience with ETA updates. 

Do you think this is excessive?

Reply:
I think that with your explanation it makes more sense. Let's see how this works IRL. We may want to tweak this later